msgraph/serviceprincipals.go

```diff
@@ -778,3 +778,135 @@ func (c *ServicePrincipalsClient) AssignAppRoleForResource(ctx context.Context,
 
 	return &appRoleAssignment, status, nil
 }
+
+// AssignTokenIssuancePolicy assigns tokenIssuancePolicies to a service principal
+func (c *ServicePrincipalsClient) AssignTokenIssuancePolicy(ctx context.Context, servicePrincipal *ServicePrincipal) (int, error) {
```

Comment:
@manicminer 
As you have said in https://github.com/manicminer/hamilton/pull/215#discussion_r1104378909, I have made the interface like `AssignClaimsMappingPolicy`.

However, I would like to be sure that my current changes are not a problem for the overall design policy of `ServicePrincipalClient`, as it differs from [your opinion](https://github.com/manicminer/hamilton/pull/215#discussion_r1112971782).

Reply:
@manicminer 

> Thanks for updating. As per my earlier comment, I think it would be best to use a function signature like

OK, I understand.

> Secondly, for the question of whether this operation should happen for applications or service principals, I note that the docs only seem to cover the `/applications/{id}/tokenIssuancePolicies` endpoint - could it be that these are applicable to both `/applications/{id}/tokenIssuancePolicies and /servicePrincipals/{id}/tokenIssuancePolicies`?

I have tried to apply a policy to the application as follows.

```sh
curl 'https://graph.microsoft.com/v1.0/applications/a191c3d6-9371-446a-a4aa-647226be2f1b/tokenIssuancePolicies/$ref' \
  -H 'Authorization: Bearer xxxxx' \
  --data-raw '{"@odata.id":"https://graph.microsoft.com/v1.0/policies/tokenIssuancePolicies/8efb6001-1369-4c68-be2d-04182c8edd76"}'
```

```json
{
    "error": {
        "code": "Request_BadRequest",
        "message": "Policy operations on v2 application are disabled.",
        "innerError": {
            "date": "2023-03-02T14:55:42",
            "request-id": "3ce11a1a-d63a-4100-9bf8-c83ca1666fbd",
            "client-request-id": "663c72ad-ccc3-2eef-38aa-3ea17b11a44b"
        }
    }
}
```

I have found that this request results in the error.
Therefore, I chose not to implement it for the application.